File: knapsack.py
class Item:
    def __init__(self, value, weight):
        self.value = value
        self.weight = weight
        self.ratio = value / weight  # Value-to-weight ratio

def fractional_knapsack(capacity, items):
    # Sort items by their value-to-weight ratio in descending order
    items.sort(key=lambda x: x.ratio, reverse=True)

    total_value = 0
    for item in items:
        if capacity == 0:
            break
        
        # Take the minimum of the remaining capacity and the current item's weight
        take_weight = min(capacity, item.weight)
        total_value += take_weight * item.ratio
        capacity -= take_weight

    return total_value

def main():
    n = int(input("Enter the number of items: "))

    items = []
    for _ in range(n):
        value = float(input("Enter the value of item: "))
        weight = float(input("Enter the weight of item: "))
        items.append(Item(value, weight))

    capacity = float(input("Enter the capacity of the knapsack: "))

    max_value = fractional_knapsack(capacity, items)
    print(f"Maximum value in the knapsack: {max_value:.2f}")

if __name__ == "__main__":
    main()

File: knapsack.java
public class knapsack {
    public static void main(String[] args) {
        int[] profit = {10,15,40};
        int[] weight = {1,2,3};
        int capacity = 6;
        int ans = getMax(profit,weight,capacity);
        System.out.println(ans);
    }
    public static int getMax(int[] profit, int[] weight, int capacity){
        int[][] dp = new int[weight.length+1][capacity+1];
        for(int i = 1 ; i <= weight.length ; i++){
            for(int j = 1 ;j <= capacity ;j++){
                if(weight[i-1]<=j){
                    dp[i][j]=Math.max(dp[i-1][j],dp[i-1][j-weight[i-1]]+profit[i-1]);
                }
                else{
                    dp[i][j] = dp[i-1][j];
                }
            }
        }
        return dp[weight.length][capacity];
    }
}